test/views/after_auth_screens/profile/edit_profile_page_test.dart

```diff
@@ -0,0 +1,226 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:talawa/view_model/after_auth_view_models/profile_view_models/edit_profile_view_model.dart';
+import 'package:talawa/views/after_auth_screens/profile/edit_profile_page.dart'; // Import the EditProfilePage widget
+
+/// Mock implementation of the [EditProfilePageViewModel] for testing purposes.
+class MockEditProfileViewModel extends Mock
+    implements EditProfilePageViewModel {}
+
+void main() {
+  group(
+    'EditProfilePage Widget Tests',
+    () {
+      testWidgets(
+        'Widget renders and updates correctly',
+        (WidgetTester tester) async {
+          final mockEditProfileViewModel = MockEditProfileViewModel();
+
+          when(mockEditProfileViewModel.initialize()).thenAnswer((_) async {});
```

Comment:
You've done this mock injection too many times. This violates the DRY principle and should be avoided. Do all of the injection and initialization during setupAll()

Reply:
Okay